src/components/Header/Header.tsx

```diff
@@ -0,0 +1,53 @@
+import classNames from 'classnames';
+
+import React from 'react';
+
+type Props = {
+  title: string,
+  setTitle: (value: string) => void,
+  onSubmit: (event: React.FormEvent) => void,
+  isDisabled: boolean,
+  toggleAll: () => void,
+  noCompletedTodos: boolean,
+};
+
+export const Header: React.FC<Props> = React.memo(
+  ({
+    title,
+    setTitle,
+    onSubmit,
+    isDisabled,
+    toggleAll,
+    noCompletedTodos,
+  }) => {
+    return (
+      <header className="todoapp__header">
+        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+        <button
+          type="button"
+          className={classNames(
+            'todoapp__toggle-all',
```

Comment:
you should save all loading todo ids to show `loader` for each todo during handling `toggleAll`, `clearCompleated`:
![image](https://user-images.githubusercontent.com/86923089/224533481-c9d0bbfc-ba94-4f08-a4f3-94589d8d028b.png)